src/Workspaces/CSharp/Portable/Recommendations/CSharpRecommendationService.cs

```diff
@@ -263,7 +263,10 @@ internal class CSharpRecommendationService : AbstractRecommendationService
                 : context.SemanticModel.LookupSymbols(context.LeftToken.SpanStart);
 
             // Filter out any extension methods that might be imported by a using static directive.
-            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod());
+            // But include extension methods declared in the context's type
+            var contextOuterTypes = context.GetOuterTypes(true, cancellationToken);
+            symbols = symbols.Where(symbol => !symbol.IsExtensionMethod()
+                || contextOuterTypes.Any(x => x.CompareTo(symbol.ContainingType) == 0));
```

Comment:
I'd prefer `outerType` to `x` for a lambda parameter here. Also, why are you using `CompareTo` to test for equality?

```C#
contextOuterTypes.Any(outerType => outerType.Equals(symbol.ContainingType));
```